File: src/main/Java/boj_step15/boj1735.java
import java.io.*;
import java.util.StringTokenizer;

public class boj1735 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int firstNumerator = Integer.parseInt(tokenizer.nextToken());
        int firstDenominator = Integer.parseInt(tokenizer.nextToken());
        tokenizer = new StringTokenizer(reader.readLine());
        int secondNumerator = Integer.parseInt(tokenizer.nextToken());
        int secondDenominator = Integer.parseInt(tokenizer.nextToken());
        int resultNumerator = firstNumerator * secondDenominator + secondNumerator * firstDenominator;
        int resultDenaminator = firstDenominator * secondDenominator;
        int temp = resultNumerator;
        resultNumerator /= gcd(resultNumerator, resultDenaminator);
        resultDenaminator /= gcd(temp, resultDenaminator);
        writer.append(resultNumerator + " " + resultDenaminator).close();
    }
    private static int gcd (int a, int b) { if (b != 0) return gcd (b, a % b); else return a; }
}

File: src/main/kotlin/boj_step15/boj1735.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
        val r = BufferedReader(InputStreamReader(System.`in`))
        val w = BufferedWriter(OutputStreamWriter(System.out))
        val (a, b) = r.readLine().split(" ").map { it.toInt() }
        val (x, y) = r.readLine().split(" ").map { it.toInt() }
        val numerator = (a * (lcm(b, y) / b)) + (x * (lcm(b, y) / y))
        val denominator = lcm(b, y)
        w.write(
                        "${numerator / gcd(numerator, denominator)} ${denominator / gcd(numerator, denominator)}"
        )
        w.close()
}

private fun gcd(a: Int, b: Int): Int = if (b != 0) gcd(b, a % b) else a

private fun lcm(a: Int, b: Int): Int = (a * b) / gcd(a, b)

